dopamine/replay_memory/circular_replay_buffer.py

```diff
@@ -109,7 +110,8 @@ def __init__(self,
                action_shape=(),
                action_dtype=np.int32,
                reward_shape=(),
-               reward_dtype=np.float32):
+               reward_dtype=np.float32,
+               lock=True):
```

Comment:
I think it's ok to turn on locking and make it non-optional. But I do like the flexibility and dependency injection that comes from having an actual lock is passed.

Reply:
As you've designed it, passing `lock=None` is an easy way to disable locking, should anyone ever want to (and I'm not sure why/when anyone would).

Reply:
I like it.